DOC-9351 Docs for Metric export to Azure Monitor supported for Dedicated Azure clusters (#18578)
cockroachdb/docs: src/current/cockroachcloud/export-metrics.md

Exporting Metrics to Azure Monitor from a CockroachDB {{ site.data.products.dedicated }} cluster is in **[limited access](https://www.cockroachlabs.com/docs/{{ site.current_cloud_version }}/cockroachdb-feature-availability)** and is only available to enrolled organizations. To enroll your organization, contact your Cockroach Labs account team. This feature is subject to change.
{{site.data.alerts.end}}

> Exporting metrics to Azure Monitor is only available on CockroachDB {{ site.data.products.dedicated }} clusters that are hosted on Azure. If your CockroachDB {{ site.data.products.dedicated }} cluster is hosted on AWS, you can export metrics to [AWS CloudWatch]({% link cockroachcloud/export-metrics.md %}?filters=aws-metrics-export). If your CockroachDB {{ site.data.products.dedicated }} cluster is hosted on GCP, you can export metrics to [Datadog]({% link cockroachcloud/export-metrics.md %}?filters=datadog-metrics-export) or [Prometheus]({% link cockroachcloud/export-metrics.md %}?filters=prometheus-metrics-export) instead.

Reviewer: ```suggestion
Exporting metrics to Azure Monitor is available only on CockroachDB {{ site.data.products.dedicated }} clusters that are hosted on Azure. If your CockroachDB {{ site.data.products.dedicated }} cluster is hosted on AWS, you can export metrics to [AWS CloudWatch]({% link cockroachcloud/export-metrics.md %}?filters=aws-metrics-export). If your CockroachDB {{ site.data.products.dedicated }} cluster is hosted on GCP, you can export metrics to [Datadog]({% link cockroachcloud/export-metrics.md %}?filters=datadog-metrics-export) or [Prometheus]({% link cockroachcloud/export-metrics.md %}?filters=prometheus-metrics-export) instead.
```
Nit

> Perform the following steps to enable metrics export from your CockroachDB {{ site.data.products.dedicated }} cluster to Azure Monitor.

Reviewer: ```suggestion
To enable metrics export to Azure Monitor:
```
A bit more streamlined and avoids duplicating the same info in line 341.

Author: updated

> 1. Find your CockroachDB {{ site.data.products.dedicated }} cluster ID that will be used in Step 5:

Reviewer: ```suggestion
1. Find your CockroachDB {{ site.data.products.dedicated }} cluster ID, which will be used in Step 5:
```
It may be difficult to remember to rename the references to other steps. I wonder if it is worth linking to the relevant step instead, then if step 5 becomes step 6, `make linkcheck` will report a broken anchor. There are a few more instances below.

	1. Visit the CockroachDB {{ site.data.products.cloud }} console [cluster page](https://cockroachlabs.cloud/clusters).
	1. Click on the name of your cluster.
	1. Find your cluster ID in the URL of the single cluster overview page: `https://cockroachlabs.cloud/cluster/{your_cluster_id}/overview`. The ID should resemble `f78b7feb-b6cf-4396-9d7f-494982d7d81e`.

> 1. Create a private key and certificate pair, using a tool such as OpenSSL. Refer to [this example]({% link {{site.current_cloud_version}}/create-security-certificates-openssl.md %}#step-1-create-the-ca-key-and-certificate-pair) to create a private key file (`ca.key`) and a certificate file (`ca.crt`) that will be uploaded in Step 3.

Reviewer: ```suggestion
1. Create a private key and certificate pair, using a tool such as OpenSSL. Refer to [Create the CA key and certificate pair]({% link {{site.current_cloud_version}}/create-security-certificates-openssl.md %}#step-1-create-the-ca-key-and-certificate-pair). This pair will be uploaded in step 3. Name the private key file (`ca.key`) and name the certificate file (`ca.crt`).
```

Author: Updated step, but not as suggested because the suggestion is not correct. Only the certificate file is uploaded in step 3.

>   1. Combine the certificate and key files using the following command. The string in the combined file will be used in Step 5.

Reviewer: ```suggestion
  1. Concatenate the certificate and key files using the following command. The concatenated file will be used in Step 5.
```

Author: updated

    {% include_cached copy-clipboard.html %}
    ~~~shell
    cat ca.crt ca.key | awk '{printf "%s\\n", $0}'  > combined.pem
    ~~~

> 1. [Create an Entra application in Azure](https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal). When registering the application, leave the optional **Redirect URI** empty since certificate-based authentication is being used, not browser-based authentication.

Reviewer: ```suggestion
1. [Create a Microsoft Entra application and service principal](https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal) in your Azure tenant. When registering the application, leave the optional **Redirect URI** empty, since certificate-based authentication is being used, not browser-based authentication.
```

Author: updated

>   1. Once the application is registered, upload the certificate file created in Step 2.

Reviewer 2: NIT: We should probably convey that `ca.crt` should be used and not `combined.pem`

Author: updated

>   1. From the **Overview** page for the Entra application, copy the values for *Display Name*, *Application (client) ID*, and the *Directory (tenant) ID* to be used in Step 5.

Reviewer: ```suggestion
  1. From the **Overview** page for the Entra application, make a note of the values for *Display Name*, *Application (client) ID*, and the *Directory (tenant) ID*, which will be used in Step 5.
```

If you choose not to link the step references, I noticed that some of them are capitalized and others are not.

Author: good catch! I made all instances of "step" lowercase.

1. [Create an Azure Monitor Application Insights resource](https://learn.microsoft.com/en-us/azure/azure-monitor/app/create-workspace-resource).

>   1. In the newly-created Application Insights resource, add a role assignment that assigns the Entra application (search for *Display Name* from Step 3) to the role [Monitoring Metrics Publisher](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/monitor#monitoring-metrics-publisher).

Reviewer: ```suggestion
  1. In the newly-created Application Insights resource, add a role assignment that assigns the Entra application (search for the *Display Name* from Step 3) to the role [Monitoring Metrics Publisher](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/monitor#monitoring-metrics-publisher).
```

Author: updated

>   1. From the **Overview** page for the Application Insights resource, from the *Connection String*, copy the values for the *InstrumentationKey* and *IngestionEndpoint* to be used in Step 5.

Reviewer: ```suggestion
  1. From the **Overview** page for the Application Insights resource, view the *Connection String*. Copy the values for *InstrumentationKey* and *IngestionEndpoint*, which will be used in Step 5.
```
I'm wondering why these are italicized?

Author: updated. I changed the italics to bold because they are UI elements.

> 1. Issue the following [Cloud API]({% link cockroachcloud/cloud-api.md %}) command to enable metrics export for your CockroachDB {{ site.data.products.dedicated }} cluster:

Reviewer: ```suggestion
1. To enable metrics export for your CockroachDB  {{ site.data.products.dedicated }} cluster, issue the following [CockroachDB Cloud API]({% link cockroachcloud/cloud-api.md %}) command:
```

Author: updated

    - `{secret_key}` is your CockroachDB {{ site.data.products.dedicated }} API key. See [API Access]({% link cockroachcloud/managing-access.md %}) for instructions on generating this key.
    - `{application_id}` is the ID of the Entra application, *Application (client) ID* from step 3.
    - `{tenant_id}` is the ID of the tenant where the Entra application was created, *Directory (tenant) ID* from step 3.
    - `{ingestion_endpoint}` is the target host where the metrics are to be sent, *IngestionEndpoint* from step 4.
    - `{instrumentation_key}` is the ID of the Azure Application Insights resource, *InstrumentationKey* from step 4.
    - `{combined_certificate_and_key}` is the string in the combined file from step 2.

1. Depending on the size of your cluster and how many regions it spans, the configuration may take a moment. You can monitor the ongoing status of the configuration using the following Cloud API command:

    {% include_cached copy-clipboard.html %}
    ~~~shell
    curl --request GET \
      --url https://cockroachlabs.cloud/api/v1/clusters/{cluster_id}/metricexport/azuremonitor \
      --header "Authorization: Bearer {secret_key}"
    ~~~

>     Run the command periodically until the command returns a status of `ENABLED`, at which point the configuration across all nodes is complete, and metrics will begin appearing in Azure Monitor. Since the configuration is applied to cluster nodes in a rolling fashion, you may see some metrics appear even before the `GET` command returns an `ENABLED` status.

Reviewer: ```suggestion
    When the command returns a status of `ENABLED`, the configuration has been applied to all nodes, and metrics will begin appearing in Azure Monitor. Since the configuration is applied to cluster nodes one at a time, metrics may begin to appear even before he status is `ENABLED`.
```

Author: updated

> 1. Once metrics export has been enabled, you can access metrics from your CockroachDB {{ site.data.products.dedicated }} cluster directly in [Azure Monitor](https://learn.microsoft.com/en-us/azure/azure-monitor/).

Reviewer: ```suggestion
Once metrics export has been enabled, you can access metrics from your CockroachDB {{ site.data.products.dedicated }} cluster directly in [Azure Monitor](https://learn.microsoft.com/en-us/azure/azure-monitor/).
```
This isn't really a step, but the procedure conclusion.

Author: updated
